src/elections/models.py

```diff
@@ -0,0 +1,62 @@
+from datetime import datetime
+
+from constants import (
+    COMPUTING_ID_LEN,
+    DISCORD_ID_LEN,
+    DISCORD_NAME_LEN,
+    DISCORD_NICKNAME_LEN,
+)
+from database import Base
+from sqlalchemy import (
+    Column,
+    DateTime,
+    ForeignKeyConstraint,
+    PrimaryKeyConstraint,
+    String,
+    Text,
+)
+
+
+# Each row represents an instance of an election
+class Election(Base):
+    __tablename__ = "election"
+
+    # Slugs are unique identifiers
+    slug = Column(String(32), primary_key=True)
+    name = Column(String(32), nullable=False)
+    # Can be one of (general_election: General Election, by_election: By-Election, council_rep_election: Council Rep Election)
+    type = Column(String(64), default="general_election")
+    date = Column(DateTime, default=datetime.now())
+    end_date = Column(DateTime)
+    websurvey = Column(String(300))
+
+# Each row represents a nominee of a given election
+class Nominee(Base):
+    __tablename__ = "election_nominee"
+
+    # Previously named sfuid
+    computing_id = Column(String(COMPUTING_ID_LEN), primary_key=True)
+    full_name = Column(String(64), nullable=False)
+    facebook = Column(String(64))
+    instagram = Column(String(64))
+    email = Column(String(64))
+    discord = Column(String(DISCORD_NAME_LEN))
+    discord_id = Column(String(DISCORD_ID_LEN))
+    discord_username = Column(String(DISCORD_NICKNAME_LEN))
+
+class NomineeSpeech(Base):
+    __tablename__ = "nominee_speech"
+
+    computing_id = Column(String(COMPUTING_ID_LEN), primary_key=True)
+    nominee_election = Column(String(32), primary_key=True)
+    speech = Column(Text)
+
+    __table_args__ = (
+        PrimaryKeyConstraint(computing_id, nominee_election),
+        ForeignKeyConstraint(
+            ["computing_id"], ["election_nominee.computing_id"]
+        ),
+        ForeignKeyConstraint(
+            ["nominee_election"], ["election.slug"]
+        )
```

Comment:
There's a slightly nicer way to mark foreign keys, which we might want to use in the future, via `ForeignKey`. Though, I'm not entirely sure, so just FYI.

The reason is that we can create an automated naming convention, so we can't accidentally name constraints incorrectly. Take a quick look at https://alembic.sqlalchemy.org/en/latest/naming.html#:~:text=automated%20naming%20convention I'll add support for this soonish!

Reply:
Can we use ForeignKey for compound fks?

Reply:
I've decided automated naming convention will be confusing & error prone. Let's just name them ourselves \:) (or write a module if we really care)